chain/chain/src/resharding/manager.rs

```diff
@@ -265,45 +229,47 @@ impl ReshardingManager {
                 retain_mode,
             )?;
 
+            // Split the parent trie and create a new child trie.
+            let trie_changes =
+                parent_trie.retain_split_shard(&split_shard_event.boundary_account, retain_mode)?;
+            let new_root = tries.apply_all(&trie_changes, parent_shard_uid, &mut store_update);
+            tries.apply_memtrie_changes(&trie_changes, parent_shard_uid, block_height);
```

Comment:
Life is good as we don't need to directly interact with memtries any more and can use the broader trie interface and `apply_all` function etc.